File: 2026POTD/JANUARY/19-01-2026/1292.ts
function maxSideLength(mat: number[][], threshold: number): number {
    const m = mat.length;
    const n = mat[0].length;
    const prefix: number[][] = Array.from({ length: m + 1 }, () => Array(n + 1).fill(0));

    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            prefix[i + 1][j + 1] = mat[i][j] + prefix[i][j + 1] + prefix[i + 1][j] - prefix[i][j];
        }
    }

    let left = 0, right = Math.min(m, n), ans = 0;

    while (left <= right) {
        const mid = Math.floor((left + right) / 2);
        let found = false;

        for (let i = mid; i <= m && !found; i++) {
            for (let j = mid; j <= n && !found; j++) {
                const total = prefix[i][j] - prefix[i - mid][j] - prefix[i][j - mid] + prefix[i - mid][j - mid];
                if (total <= threshold) found = true;
            }
        }

        if (found) {
            ans = mid;
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return ans;
}

File: 2026POTD/JANUARY/19-01-2026/1292.swift
class Solution {
    func maxSideLength(_ mat: [[Int]], _ threshold: Int) -> Int {
        let m = mat.count
        let n = mat[0].count
        
        var prefix = Array(repeating: Array(repeating: 0, count: n + 1), count: m + 1)
        for i in 0..<m {
            for j in 0..<n {
                prefix[i + 1][j + 1] = mat[i][j] + prefix[i][j + 1] + prefix[i + 1][j] - prefix[i][j]
            }
        }
        
        var left = 0
        var right = min(m, n)
        var ans = 0
        
        while left <= right {
            let mid = (left + right) / 2
            var found = false
            
            outer: for i in mid...m {
                for j in mid...n {
                    let total = prefix[i][j] - prefix[i - mid][j] - prefix[i][j - mid] + prefix[i - mid][j - mid]
                    if total <= threshold {
                        found = true
                        break outer
                    }
                }
            }
            
            if found {
                ans = mid
                left = mid + 1
            } else {
                right = mid - 1
            }
        }
        
        return ans
    }
}